src/meta/src/stream/fragmenter.rs

```diff
@@ -141,9 +147,64 @@ where
             .collect::<Result<BTreeMap<_, _>>>()
     }
 
+    /// Do some dirty rewrites on meta. Currently, it will split stateful operators into two
+    /// fragments.
+    fn rewrite_stream_node(&self, stream_node: StreamNode) -> Result<StreamNode> {
+        self.rewrite_stream_node_inner(stream_node, false)
+    }
+
+    fn rewrite_stream_node_inner(
```

Comment:
Can `rewrite_stream_node_inner` be done in frontend? It might be more reasonable. 🤔

Reply:
Then we will have to move the whole fragmenter to frontend, and having a proto to pass fragment among nodes 🤣 Currently, fragments are only intermediate representation that are generated on meta node.